evap/results/exporters.py

```diff
@@ -79,220 +87,193 @@ def filter_text_and_heading_questions(questions):
+
+    def write_headings_and_evaluation_info(self, evaluations_with_results, semesters, contributor, degrees, course_types):
+        export_name = "Evaluation"
+        if contributor:
+            export_name += "\n{}".format(contributor.full_name)
+        elif len(semesters) == 1:
+            export_name += "\n{}".format(semesters[0].name)
+        degree_names = [degree.name for degree in Degree.objects.filter(pk__in=degrees)]
+        course_type_names = [course_type.name for course_type in CourseType.objects.filter(pk__in=course_types)]
+        self.write_cell(_("{}\n\n{}\n\n{}").format(
+            export_name, ", ".join(degree_names), ", ".join(course_type_names)
+        ), "headline")
+
+        for evaluation, __ in evaluations_with_results:
+            title = evaluation.full_name
+            if len(semesters) > 1:
+                title += "\n{}".format(evaluation.course.semester.name)
+            responsible_names = [responsible.full_name for responsible in evaluation.course.responsibles.all()]
+            title += "\n{}".format(", ".join(responsible_names))
+            self.write_cell(title, "evaluation")
+
+        self.next_row()
+        self.write_cell(_("Degrees"), "bold")
+        for evaluation, __ in evaluations_with_results:
+            self.write_cell("\n".join([d.name for d in evaluation.course.degrees.all()]), "degree")
+
+        self.next_row()
+        self.write_cell(_("Course Type"), "bold")
+        for evaluation, __ in evaluations_with_results:
+            self.write_cell(evaluation.course.type.name, "border_left_right")
+
+        self.next_row()
+        self.write_empty_row_with_styles(["border_left_right"] * len(evaluations_with_results))
+
+    def write_overall_results(self, evaluations_with_results, course_results_exist):
+        evaluations = list(map(lambda t: t[0], evaluations_with_results))
+
+        self.write_cell(_("Overall Average Grade"), "bold")
+        averages = map(lambda e: distribution_to_grade(calculate_average_distribution(e)), evaluations)
+        self.write_row(averages, lambda avg: self.grade_to_style(avg) if avg else "border_left_right")
+
+        self.write_cell(_("Total voters/Total participants"), "bold")
+        voter_ratios = map(lambda e: f"{e.num_voters}/{e.num_participants}", evaluations)
+        self.write_row(voter_ratios, style="total_voters")
+
+        self.write_cell(_("Evaluation rate"), "bold")
+        # round down like in progress bar
+        participant_percentages = map(lambda e: int((e.num_voters / e.num_participants) * 100) if e.num_participants > 0 else 0, evaluations)
+        participant_percentages = map(lambda p: f"{p}%", participant_percentages)
+        self.write_row(participant_percentages, style="evaluation_rate")
+
+        if course_results_exist:
+            # Only query the number of evaluations once and keep track of it here.
+            count_gt_1 = list(map(lambda e: e.course_evaluations_count > 1, evaluations))
```

Comment:
have you considered using list comprehensions here and in 180?

Reply:
I haven't, good idea. I removed all occurrences of `map` in the class and replaced them with list or generator expressions. :+1: